aptos-move/block-executor/src/scheduler.rs

```diff
@@ -247,15 +247,16 @@ impl Scheduler {
                     ))
                 })
                 .collect(),
-            commit_state: CachePadded::new(Mutex::new((0, 0))),
-            execution_idx: AtomicU32::new(0),
-            validation_idx: AtomicU64::new(0),
+            commit_state: CachePadded::new(Mutex::new((first_index, 0))),
+            execution_idx: AtomicU32::new(first_index),
+            validation_idx: AtomicU64::new(first_index as u64),
             done_marker: CachePadded::new(AtomicBool::new(false)),
         }
     }
 
-    pub fn num_txns(&self) -> TxnIndex {
-        self.num_txns
+    pub fn is_last_index(&self, index: TxnIndex) -> bool {
+        index != self.index_mapping.end_index
```

Comment:
Seem my suggestion in the code, you can just remove this function with that helper function I guess.

Reply:
This function is needed by `Scheduler::coordinator_commit_hook` though.

Reply:
didn't realize it was `Scheduler::is_last_index()`. fixed.